File: official(c)/Junior/j4.java
import java.util.Scanner;

public class j4 {
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        
        int N = input.nextInt();
        char[] days = new char[N];
        for (int i = 0; i < N; i++) {
            days[i] = input.next().charAt(0);
        }
        
        
        int[] left = new int[N];
        left[0] = (days[0] == 'S' ? 1 : 0);
        for (int i = 1; i < N; i++) {
            left[i] = (days[i] == 'S' ? left[i-1] + 1 : 0);
        }
        
        
        int[] right = new int[N];
        right[N-1] = (days[N-1] == 'S' ? 1 : 0);
        for (int i = N-2; i >= 0; i--) {
            right[i] = (days[i] == 'S' ? right[i+1] + 1 : 0);
        }
        
        
        int ans = 0;
        boolean hasP = false;
        for (int i = 0; i < N; i++) {
            if (days[i] == 'P') {
                hasP = true;
                int leftCount = (i > 0 ? left[i-1] : 0);
                int rightCount = (i < N-1 ? right[i+1] : 0);
                ans = Math.max(ans, leftCount + 1 + rightCount);
            }
        }
        
        if (!hasP) {
            ans = N - 1;
        }
        
        System.out.println(ans);
        input.close();
    }
}

File: official(c)/Junior/j4.py
N = int(input())
days = list(input().split())

left = [0] * N
left[0] = 1 if days[0] == 'S' else 0
for i in range(1, N):
    left[i] = left[i-1] + 1 if days[i] == 'S' else 0

right = [0] * N
right[N-1] = 1 if days[N-1] == 'S' else 0
for i in range(N-2, -1, -1):
    right[i] = right[i+1] + 1 if days[i] == 'S' else 0

ans = 0
has_p = False
for i in range(N):
    if days[i] == 'P':
        has_p = True
        left_count = left[i-1] if i > 0 else 0
        right_count = right[i+1] if i < N-1 else 0
        ans = max(ans, left_count + 1 + right_count)

if not has_p:
    ans = N - 1

print(ans)